src/veh_interact.cpp

```diff
@@ -1357,7 +1370,19 @@ bool veh_interact::can_remove_part( int idx ) {
         qual = JACK;
         lvl = jack_qality( *veh );
         str = veh->lift_strength();
-        use_aid = max_jack >= lvl;
+
+        auto self_jacking_parts = veh->get_parts( "SELF_JACK", false );
+        bool can_self_jack = false;
+        for( auto jack : self_jacking_parts )
+        {
+            item jack_item( jack->properties_to_item() );
+            if( jack_item.has_quality( SELF_JACK, lvl ) ) {
+                can_self_jack = true;
+                break;
+            }
+        }
```

Comment:
This could be rewritten as:

```cpp
        bool can_self_jack = std::any_of(self_jacking_parts.cbegin(), self_jacking_parts.cend(), [lvl](auto jack)
        {
            return jack->properties_to_item().has_quality( SELF_JACK, lvl );
        });
```

Obviously this better as:

```cpp
bool veh_interact::can_self_jack() {
        auto self_jacking_parts = veh->get_parts( "SELF_JACK", false );
        int lvl = jack_qality( *veh );
        return std::any_of(self_jacking_parts.cbegin(), self_jacking_parts.cend(), [lvl](auto jack)
        {
            return jack->properties_to_item().has_quality( SELF_JACK, lvl );
        });
```

Note: I didn't compile these so they may have syntax errors and such.